tests/ext-popup.test.js

```diff
@@ -0,0 +1,48 @@
+import Popup from '../src/lib/ext-popup';
+import * as ExtListen from '../src/lib/ext-listen';
+
+test('test init popup when no extension is being monitored', async () => {
+  document.body.innerHTML =
+    '<div>' +
+    '  <p id="monitorStatus"></p>' +
+    '  <button id="startMonitorBtn"></button>' +
+    '  <button id="stopMonitorBtn"></button>' +
+    '  <button id="actLogPage"></button>' +
+    '  <p id="errorText"></p>' +
+    '</div>';
+
+  const areExtsBeingMonitoredFn = jest.spyOn(
+    ExtListen,
+    'areExtsBeingMonitored'
+  );
+
+  const startBtn = document.getElementById('startMonitorBtn');
+  const stopBtn = document.getElementById('stopMonitorBtn');
+  const monitorStatus = document.getElementById('monitorStatus');
+  const errorText = document.getElementById('errorText');
+
+  const popup = new Popup();
+
+  const renderMonitorStartedUIFn = jest.spyOn(popup, 'renderMonitorStartedUI');
+  const addListenerPopupFn = jest.spyOn(popup, 'addListenerPopup');
+
+  areExtsBeingMonitoredFn.mockImplementation(() => {
+    return new Promise((resolve) => resolve(true));
+  });
+
+  await popup.init();
+  expect(addListenerPopupFn).toBeCalledTimes(3);
+  expect(renderMonitorStartedUIFn).toHaveBeenCalledTimes(1);
```

Comment:
I don't think that we are not that much interested into how many times we have called this function, as much as we are interested in checking that the status of the elements in the popup is the expected ones (which is being checked below).

In my opinion we can get read of these two assertions.

Reply:
Yeah, deleted this two assertions.